shared/src/main/kotlin/com/egm/stellio/shared/web/CustomWebFilter.kt

```diff
@@ -0,0 +1,44 @@
+package com.egm.stellio.shared.web
+
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.stereotype.Component
+import org.springframework.web.server.ResponseStatusException
+import org.springframework.web.server.ServerWebExchange
+import org.springframework.web.server.WebFilter
+import org.springframework.web.server.WebFilterChain
+import reactor.core.publisher.Mono
+
+@Component
+class CustomWebFilter : WebFilter {
+
+    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
+        val request: ServerHttpRequest = exchange.request
+        return when (request.method) {
+            HttpMethod.POST -> httpPostRequestPreconditions(exchange, chain)
+            HttpMethod.PATCH -> httpPatchRequestPreconditions(exchange, chain)
+            else -> chain.filter(exchange)
+        }
+    }
+
+    private fun httpPostRequestPreconditions(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
+        val request: ServerHttpRequest = exchange.request
+        val contentLength = request.headers.contentLength
```

Comment:
you are doing twice the same check for content-length header, move that in a method that will do the check